cli/command_snapshot_create.go

```diff
@@ -69,6 +70,7 @@ func (c *commandSnapshotCreate) setup(svc appServices, parent commandParent) {
 	cmd.Flag("tags", "Tags applied on the snapshot. Must be provided in the <key>:<value> format.").StringsVar(&c.snapshotCreateTags)
 	cmd.Flag("pin", "Create a pinned snapshot that's will not expire automatically").StringsVar(&c.pins)
 	cmd.Flag("flush-per-source", "Flush writes at the end of each source").Hidden().BoolVar(&c.flushPerSource)
+	pathVar(cmd.Flag("snapshot-path", "Directory or file that should be snapshot instead of source."), &c.snapshotPath)
```

Comment:
Decided to go this way instead of allowing to override the source that will be stored in the repo for more consistency with --stdin-file

Reply:
An interesting alternative would be to provide a --set-source flag as suggested in the issue originally that bypasses path validation to some degree.
One usecase I would see for this is that it would allow to create snapshots of a windows path from a unix system and the other way around. Disadvantage here is that we need to decide what exactly a valid path is for us and get rid of all the platform specific path handling for sourceinfo.

wdyt?